client.go

```diff
@@ -0,0 +1,331 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// tokenExpiredError Error
+type tokenExpiredError struct {
+}
+
+func (e tokenExpiredError) Error() string {
+	return "Expired Token"
+}
+
+// NotFound Error
+type NotFound struct {
+}
+
+func (e NotFound) Error() string {
+	return "Token Not Found"
+}
+
+// VisualizationError errors for Visualization client
+type VisualizationError struct {
+	code        string
+	message     string
+	description string
+}
+
+// Error generate a error message.
+// If Code is zero, we know it's not a http error.
+func (e VisualizationError) Error() string {
+	return fmt.Sprintf("ERROR: %s", e.description)
+}
+
+// VisualizationClient client for Visualization
+type VisualizationClient struct {
+	url            string
+	client         *http.Client
+	token          AuthToken
+	openstackToken string
+}
+
+// NewVisualizationClient returns client with token
+func NewVisualizationClient(url string, token AuthToken) (*VisualizationClient, error) {
+	client := http.Client{}
+	return &VisualizationClient{client: &client, url: url, token: token}, nil
+
+}
+
+// A JSONMessage contains the json error message received when http request failed
+type JSONMessage struct {
+	Message string `json:"message"`
+}
+
+func (v *VisualizationClient) reIssue() bool {
+	_, err := v.OpenstackAuth()
+	return err == nil
+}
+
+// authorizedHttpRequest for token checking and reauth
+func (v *VisualizationClient) authorizedHttpRequest() (err error) {
```

Comment:
Function called authorizedHttpRequest, however it does no http request actually.